### PR TITLE
feat(#29): attach reference files to bookings

### LINKED ISSUE #29: Attach Reference Files
### User story
As a user, I am able to include attachments for reference in my room booking.
### Narrative
As a user, certain room bookings may require the submission of a form, waiver or document; it should therefore be possible to attach reference files of this sort to a room booking.
### Out of scope
Validation of the reference documents in question.
### Acceptance criteria

| # | Given | When | Then |
|---|-------|------|------|
|1|I am a user in the process of booking a room|I am required to include additional reference files|I should have the option to attach files to my room booking submission|

### UI Mockup:
image

### PR DESCRIPTION
**Pull request by KeeganWoods1**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**
Please provide enough information so that others can review your pull request:

Allows users to upload files with their booking request submission. Files can be added at booking creation or later by updating an existing booking.

### Test plan (required)
Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

Tests have passed.

### Closing issues
- closes #29  

### Pre-Merge checklist 
Complete these items before requesting reviews
- [x] Static Analysis Rules Validated
- [x] All commits follow naming conventions 

### Sub-tasks Checklist
- [ ] UI Implementation
- [ ] UI Test
- [x] Backend Implementation
- [x] Backend Tests
- [x] Integration Tests
- [x] System Tests
- [ ] Documentation

### CODE REVIEW

File: resources/js/Pages/Admin/BookingRequests/UpdateBookingRequestForm.vue [hunk 1]
```diff
@@ -95,18 +104,21 @@ export default {
...
+            availableExcludingCurrent: [], 
             form: this.$inertia.form(
-                {
+                { 
                     user_id: null,
                     room_id: null,
                     start_time: null,
                     end_time: null,
+                    reference: [],
+                    _method: 'PUT'
```
Comment (jonathanp16) [inline resources/js/Pages/Admin/BookingRequests/UpdateBookingRequestForm.vue:115]: Can't seem to find where _method is being used

File: resources/js/Pages/Admin/BookingRequests/UpdateBookingRequestForm.vue [hunk 2]
```diff
@@ -118,15 +130,26 @@ export default {
...
+        fieldChange(e){
+            let selectedFiles = e.target.files;
+
+            if(!selectedFiles.length)
+                return false;
+
+            for(let i=0;i<selectedFiles.length;i++)
+            {
+                this.form.reference.push(selectedFiles[i]);
+            }
```
Comment (jonathanp16) [inline resources/js/Pages/Admin/BookingRequests/UpdateBookingRequestForm.vue:152]: Same here should use foreach

**Review by jonathanp16 — CHANGES_REQUESTED** [review]
Great job. Just some typos and some questions that I left as comments

File: resources/js/Pages/Admin/BookingRequests/CreateBookingRequestForm.vue [hunk 3]
```diff
@@ -8,7 +8,7 @@
           <div class="m-6">
             <jet-label for="room_id" value="Room" />
             <select v-model="createBookingRequestForm.room_id" class="mt-1 block w-full" name="availableRooms" id="availableRooms">
-              <option :value="null" selected="selected">Choose here</option>
+              <option :value="null" selected="selected">Chose here</option>
```
[the remark this answers is not in the document]
Comment (KeeganWoods1) [reply resources/js/Pages/Admin/BookingRequests/CreateBookingRequestForm.vue]: doone

File: resources/js/Pages/Admin/BookingRequests/CreateBookingRequestForm.vue [hunk 4]
```diff
@@ -90,21 +100,36 @@ export default {
         {
           room_id: null,
           start_time: "",
-          end_time: ""
+          end_time: "",
+          reference: [],
+          file: null,
```
[the remark this answers is not in the document]
Comment (KeeganWoods1) [reply resources/js/Pages/Admin/BookingRequests/CreateBookingRequestForm.vue]: This never should have been there, removed it.

File: resources/js/Pages/Admin/BookingRequests/CreateBookingRequestForm.vue [hunk 5]
```diff
@@ -90,21 +100,36 @@ export default {
...
+      let selectedFiles = e.target.files;
+
+      if(!selectedFiles.length)
+        return false;
+
+      for(let i=0;i<selectedFiles.length;i++)
+      {
+        this.createBookingRequestForm.reference.push(selectedFiles[i]);
+      }
+    }
```
[the remark this answers is not in the document]
Comment (KeeganWoods1) [reply resources/js/Pages/Admin/BookingRequests/CreateBookingRequestForm.vue]: Agreed

File: resources/js/Pages/Admin/BookingRequests/UpdateBookingRequestForm.vue (hunk 1 above)
Comment (KeeganWoods1) [reply resources/js/Pages/Admin/BookingRequests/UpdateBookingRequestForm.vue:115]: This is Laravel's method spoofing mechanism. In order to send an array of files to the controller via a PUT request we must pass this hidden "_method" variable through a POST.

File: resources/js/Pages/Admin/BookingRequests/UpdateBookingRequestForm.vue (hunk 2 above)
Comment (KeeganWoods1) [reply resources/js/Pages/Admin/BookingRequests/UpdateBookingRequestForm.vue]: Agreed.

**Review by EvanDime — APPROVED** [review]

**Review by jonathanp16 — APPROVED** [review]
Good work and really good coverage

**Pull request merged**